smartcontract/service/native/cross_chain/lock_proxy/lock_proxy.go

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright (C) 2018 The ontology Authors
+ * This file is part of The ontology library.
+ *
+ * The ontology is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ontology is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with The ontology.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package lock_proxy
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"github.com/ontio/ontology/common"
+	"github.com/ontio/ontology/common/config"
+	"github.com/ontio/ontology/core/genesis"
+	"github.com/ontio/ontology/core/types"
+	"github.com/ontio/ontology/smartcontract/event"
+	"github.com/ontio/ontology/smartcontract/service/native"
+	"github.com/ontio/ontology/smartcontract/service/native/cross_chain/cross_chain_manager"
+	"github.com/ontio/ontology/smartcontract/service/native/ont"
+	"github.com/ontio/ontology/smartcontract/service/native/utils"
+	"math/big"
+)
+
+func InitLockProxy() {
+	native.Contracts[utils.LockProxyContractAddress] = RegisterLockProxyContract
+}
+
+func RegisterLockProxyContract(native *native.NativeService) {
+	native.Register(LOCK_NAME, Lock)
+	native.Register(UNLOCK_NAME, Unlock)
+	native.Register(BIND_PROXY_NAME, BindProxyHash)
+	native.Register(BIND_ASSET_NAME, BindAssetHash)
+	native.Register(WITHDRAW_ONG_NAME, WithdrawONG)
+	native.Register(GET_PROXY_HASH_NAME, GetProxyHash)
+	native.Register(GET_ASSET_HASH_NAME, GetAssetHash)
+	native.Register(GET_CROSSED_AMOUNT_NAME, GetCrossedAmount)
+	native.Register(GET_CROSSED_LIMIT_NAME, GetCrossedLimit)
+}
+
+func BindProxyHash(native *native.NativeService) ([]byte, error) {
+	contract := native.ContextRef.CurrentContext().ContractAddress
+	source := common.NewZeroCopySource(native.Input)
+	var bindParam BindProxyParam
+	if err := bindParam.Deserialization(source); err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[BindProxyHash] Deserialize BindProxyParam error:%s", err)
+	}
+	operatorAddress, err := types.AddressFromBookkeepers(genesis.GenesisBookkeepers)
```

Comment:
use operator address from global param like in header sync contract.

Reply:
updated